java/build.gradle

```diff
@@ -0,0 +1,123 @@
+buildscript {
+    repositories {
+        maven { url 'https://plugins.gradle.org/m2/' }
+    }
+    dependencies {
+        classpath 'com.google.protobuf:protobuf-gradle-plugin:0.8.17'
+        classpath "org.jfrog.buildinfo:build-info-extractor-gradle:4.4.7"
+        classpath "org.ajoberstar:gradle-git:1.6.0"
+    }
+}
+
+plugins {
+    id 'java'
+    id "com.google.protobuf" version "0.9.1"
+    id 'maven-publish'
+    id 'signing'
+    id "io.github.gradle-nexus.publish-plugin" version "1.1.0"
+}
+
+group 'io.odpf'
+version '0.0.1'
+
+repositories {
+    mavenCentral()
+    jcenter()
+    mavenLocal()
+    maven {
+        url "https://plugins.gradle.org/m2/"
+    }
+}
+
+sourceSets {
+    main {
+        proto {
+            srcDir "${projectDir}/../odpf/raccoon/"
```

Comment:
if we plan to only publish racoon related schemas we should name the `groupId` and `artifactID` accordingly.

Reply:
I tried generating the other protos but they were failing, 
so for the time being, I've only included the raccoon.

Reply:
@AkbaraliShaikh +1 on this. I think we should publish all protos with proton. If it is just raccoon then we can use raccoon repo to generate code based on a proton commit. In the same way, we are doing this for go.

Reply:
I wanted to publish all the protos, not just the raccoon ones, but there was a problem with the dependent protos that were used by other protos in this repo.

In case of golang/buf we can simply provide the dependent list like below.
```
deps:
  - buf.build/googleapis/googleapis
  - buf.build/grpc-ecosystem/grpc-gateway
  - buf.build/envoyproxy/protoc-gen-validate
```

but in case of java (gradle), we might have to download all the dependent locally to generate.
It is not easy like the buf.

Reply:
We can use `buf` to take care of the java code generation instead of `protoc`. Once code is generated we can use `gradle` for packaging and publishing the jar.

Reply:
+1

Reply:
It'd be cleaner to have a multi-module project (either with gradle or buf way) where
the group can become `io.odpf.proton` (groupId) as the parent project and depends on each child's artefact project builds. The artifactId pushed is `raccoon|optimus|xxx`
This way in future, new children can built separately & bundle the way it needs to.

Reply:
If multi-module is time intensive to do now, we can adopt to move to building it in Racccon 

@AkbaraliShaikh

Reply:
+1 I think we are thinking of just compiling raccoon now, we use generate code within raccoon java client using buf itself and package into raccoon client jar. and convert this pipeline to entire proton pipeline. Later on if needed we can setup per project workflow.